soc/xtensa/intel_apl_adsp/adsp.c

```diff
@@ -78,6 +78,14 @@ static void prepare_host_windows(void)
 	SOC_DCACHE_FLUSH((void *)(HP_SRAM_WIN0_BASE + SRAM_REG_FW_END),
 			 HP_SRAM_WIN0_SIZE - SRAM_REG_FW_END);
 
+	if (IS_ENABLED(CONFIG_IPM_ADSP)) {
```

Comment:
I'm thinking long term we should build the extended FW boot data (i.e. data we send to host about window mappings and some FW capabilities via DT). Any thoughts on this, not something we need immediately but worth considering ?

Reply:
At the moment we send window mappings:
https://github.com/zephyrproject-rtos/zephyr/blob/93b78b7bf8bfad75196de969a7487f95f235aad3/soc/xtensa/intel_apl_adsp/adsp.c#L91-107
On the host side we use https://github.com/thesofproject/sof-diagnostic-driver which does not use it much.

Reply:
Linux and the diagnostic driver read this data as part of their IPC init, since mailbox sizes, access etc and IPC HW varies between the SoCs so I dont think there is anything stopping the DT upgrade when time permits.